_episodes/08-connected-components.md

```diff
@@ -261,156 +319,388 @@ max: 11
+> > histogram as a continuous curve) that tell us about certain groups
+> > in the image.
+> >
+> > In this example, we can see that there are four small objects that
+> > contain less than 50000 pixels. Then there is a group of four
+> > (1+1+2) objects in the range between 20000 and 400000, and three
+> > objects with a size around 500000. For our object count, we might
+> > want to disregard the small objects as artifacts, i.e, we want to
+> > ignore the leftmost bar of the histogram. We could use a threshold
+> > of 50000 as the minimum area to count. In fact, the `object_areas`
+> > list already tells us that there are fewer than 200 pixels in
+> > these objects. Therefore, it is reasonable to require a minimum
+> > area of at least 200 pixels for a detected object. In practice,
+> > finding the "right" threshold can be tricky and usually involves
+> > an educated guess based on domain knowledge.
 > {: .solution}
 {: .challenge}
 
-> ## Filter objects by area (15 min)
->
-> Our `CCA-count.py` program has an apparent problem:
-> it is hard to find a combination that produces the right output number.
-> In some cases the problem arose that some background noise got picked up as an object.
-> With other parameter settings some of the foreground objects got broken up or disappeared completely.
+> ## Filter objects by area (10 min)
 >
-> ![junk.jpg mask detail](../fig/09-junk-cca-detail.png)
->
-> Modify the program in order to only count large objects.
+> Now we would like to use a minimum area criterion to obtain a more
+> accurate count of the objects in the image.
 >
+> 1. Find a way to calculate the number of objects by only counting
+> objects above a certain area.
 >
 > > ## Solution
 > >
+> > One way to count only objects above a certain area is to first
+> > create a list of those objects, and then take the length of that
+> > list as the object count. This can be done as follows:
+> >
+> > ~~~
+> > min_area = 10000
+> > large_objects = []
+> > for objf in object_features:
+> >     if objf["area"] > min_area:
+> >         large_objects.append(objf["label"])
+> > print("Found", len(large_objects), "objects!")
+> > ~~~
+> > {: .language-python}
+> >
+> > Another option is to use the Numpy function `np.where` to create
+> > the list of large objects. This function can be used to access
+> > only the indices of an an array where the specified condition is
+> > True. We first create an array `object_areas` containing the
+> > object areas, and an array `object_labels` containing the object
+> > labels. The labels of the objects are also returned by
+> > `skimage.measure.regionprops`. We can then use `np.where` to
+> > select the labels of objects whose area is greater than
+> > `min_area`:
+> >
+> > ~~~
+> > object_areas = np.array([objf["area"] for objf in object_features])
+> > object_labels = np.array([objf["label"] for objf in object_features])
+> > large_objects = object_labels[np.where(object_areas > min_area)]
```

Comment:
```suggestion
> > large_objects = object_labels[object_areas > min_area]
```
I don't think you actually need ```np.where``` here. ```object_areas > min_area``` produces a boolean numpy array of the same dimensions as ```object_labels``` so it can be used directly for indexing. Perhaps it's worth removing ```np.where```, then this saves on explaining another numpy function?

Reply:
Good catch @K-Meech. `np.where` is actually an alias to `np.nonzero` here and can be shorthanded as you suggest. I'll make that change.

Reply:
Removed use of `np.where` in b878897f2f62261f6c4899459b1ee01dc67b32f4 and rephrased the explanation.